pkg/controller/deployment/deployment_controller.go

```diff
@@ -419,17 +430,52 @@ func (dc *DeploymentController) syncDeployment(key string) error {
 	return fmt.Errorf("unexpected deployment strategy type: %s", d.Spec.Strategy.Type)
 }
 
-func (dc *DeploymentController) syncRecreateDeployment(deployment extensions.Deployment) error {
-	newRC, err := dc.getNewRC(deployment)
+// Rolling back to a version; no-op if the toVersion is deployment's current version
+func (dc *DeploymentController) rollback(deployment extensions.Deployment, toVersion int) error {
+	if toVersion <= 0 {
```

Comment:
If no version is specified, I think we should rollback to the previous version (2nd largest version number).